local yarn version interferes with nix-shell

# Problem

`make run-clojure` fails with the following message

```
make run-clojure                                        
yarn shadow-cljs watch mobile
Configuring Nix shell for target 'clojure'...
Node modules not created by Nix
Copying node_modules from Nix store:
 - /nix/store/h6cvkw9rmr40ilqs0aavx0cb3dq9wgcr-status-mobile-node-deps-2.32.0-patched
Done in: 14.962s
Internal Error: StatusIm-Mobile@workspace:.: This package doesn't seem to be present in your lockfile; run "yarn install" to update the lockfile
    at V0.getCandidates (/Users/john/.yarn/releases/yarn-3.6.4.cjs:435:5145)
    at kf.getCandidates (/Users/john/.yarn/releases/yarn-3.6.4.cjs:391:1264)
    at /Users/john/.yarn/releases/yarn-3.6.4.cjs:439:8033
    at df (/Users/john/.yarn/releases/yarn-3.6.4.cjs:390:11070)
    at ge (/Users/john/.yarn/releases/yarn-3.6.4.cjs:439:8013)
    at async Promise.allSettled (index 0)
    at async io (/Users/john/.yarn/releases/yarn-3.6.4.cjs:390:10398)
    at async /Users/john/.yarn/releases/yarn-3.6.4.cjs:439:8545
    at async ti.startProgressPromise (/Users/john/.yarn/releases/yarn-3.6.4.cjs:390:47651)
    at async je.resolveEverything (/Users/john/.yarn/releases/yarn-3.6.4.cjs:439:6573)
make: *** [run-clojure] Error 1
```

after installing packages inside shell

```-prefixed code
➤ YN0000: └ Completed in 2m 3s
➤ YN0000: ┌ Fetch step
➤ YN0013: │ yargs@npm:13.3.2 can't be found in the cache and will be fetched from the remote registry
➤ YN0013: │ yargs@npm:15.4.1 can't be found in the cache and will be fetched from the remote registry
➤ YN0013: │ yargs@npm:17.6.2 can't be found in the cache and will be fetched from the remote registry
➤ YN0013: │ yargs@npm:17.7.2 can't be found in the cache and will be fetched from the remote registry
➤ YN0013: │ yocto-queue@npm:0.1.0 can't be found in the cache and will be fetched from the remote registry
➤ YN0000: └ Completed in 1s 365ms
➤ YN0000: ┌ Link step
➤ YN0007: │ nodemon@npm:2.0.16 must be built because it never has been before or the last one failed
➤ YN0007: │ bufferutil@npm:4.0.8 must be built because it never has been before or the last one failed
➤ YN0007: │ es5-ext@npm:0.10.64 must be built because it never has been before or the last one failed
➤ YN0007: │ utf-8-validate@npm:5.0.10 must be built because it never has been before or the last one failed
➤ YN0007: │ core-js@npm:2.6.12 must be built because it never has been before or the last one failed
➤ YN0007: │ StatusIm-Mobile@workspace:. must be built because it never has been before or the last one failed
➤ YN0000: └ Completed in 25s 134ms```

downgrading yarn to version 1.22.22 did not fix the issue. It's probably something to do with my dev enviroment. 

@siddarthkay

after a clean nix installation and with yarn 1.22.22

`Node modules not created by Nix
Copying node_modules from Nix store:
 - /nix/store/h6cvkw9rmr40ilqs0aavx0cb3dq9wgcr-status-mobile-node-deps-2.32.0-patched
Done in: 18.834s
error This project's package.json defines "packageManager": "yarn@4.6.0". However the current global version of Yarn is 1.22.22.

Presence of the "packageManager" field indicates that the project is meant to be used with Corepack, a tool included by default with all official Node.js distributions starting from 16.9 and 14.19.
Corepack must currently be enabled by running corepack enable in your terminal. For more information, check out https://yarnpkg.com/corepack.
make: *** [run-clojure] Error 1`

--- a/nix/shell.nix
+++ b/nix/shell.nix
@@ -38,6 +38,10 @@ in mkShell {
   LANG="en_US.UTF-8";
   LANGUAGE="en_US.UTF-8";
 
+  # Prevent Yarn from checking packageManager field in parent directories
+  # https://github.com/yarnpkg/yarn/blob/740c38c3a962c30ddb344a919bbfb7065620714b/src/cli/index.js#L292C21-L292C33
+  SKIP_YARN_COREPACK_CHECK="1";
+
   # Important to load our own Nix binary cache.
   NIX_CONFIG = builtins.readFile ./nix.conf;
 

